pkg/vault/client.go

```diff
@@ -122,3 +121,49 @@ func (vc *vaultClient) SecretVersionFromVault(secretName string) (int64, error)
 
 	return version, nil
 }
+
+// EnginesFromVault takes a path and returns a list of engines from vault.
+func (vc *vaultClient) EnginesFromVault(path string) ([]string, error) {
+	if vc.config.traceEnabled {
+		var span *trace.Span
+		vc.ctx, span = trace.StartSpan(vc.ctx, fmt.Sprintf("%s/EnginesFromVault", vc.config.tracePrefix))
+		defer span.End()
+	}
+
+	engines, err := vc.client.Logical().List(path)
+	if err != nil {
+		log.Error(fmt.Sprintf("listing engines from Vault: %v", err))
+		return nil, fmt.Errorf("listing engines from Vault for %s", path)
+	}
+
+	if engines == nil {
+		log.Error("engines returned from Vault are <nil> for " + path)
+		return nil, fmt.Errorf("engines returned from Vault are <nil> for %s", path)
+	}
+
+	engineData, _ := extractListData(engines)
+
+	result := []string{}
+
+	for key, value := range engineData {
+		fmt.Printf("key: %s, value: %v \n", key, value)
+		result = append(result, value.(string))
+	}
```

Comment:
I think this would be a great place to take advantage of a [type switch](https://tour.golang.org/methods/16).  We know we want a string in the result from vault, but the code as written would throw a runtime error because we're telling the compiler that the result from vault here will always be a string, guaranteed.  Given that we have little control over the response, the following code would be safer, and provides a useful error.

```suggestion
	for key, value := range engineData {
	        switch v := vaule.(type)
	        case string:
	            result = append(result, v)
	         default:
	             return fmt.Errorf("unexpected type, expected string, got: %T, value: %v", v, v)
	         }
  	}
```

Reply:
Thanks, Paul 👍  Added a type switch in.